### PR TITLE
[Merged by Bors] - Use extra parameters in connector smartmodule

### LINKED ISSUE #285: Support SmartModule user-defined parameters in connector configuration
I would like to be able to pass smart module parameters when configuring a connector, as described in https://www.infinyon.com/blog/2021/12/smartmodule-params/ for consumers.

This is currently not possible and it seems that you can only support default parameters when looking at https://github.com/infinyon/fluvio-connectors/blob/main/rust-connectors/common/src/opt.rs#L234

During the configuration of a connector I'm not able to pass --extra-params like when configuring a consumer with a smart module:
```yaml
version: 0.2.0
name: cat-facts
type: http
topic: cat-facts
direction: source
parameters:
  endpoint: https://fact-apis.herokuapp.com/facts/cat
  interval: 15
  map: petfact-mapper
  extra-params: about=cat
```
fails to produce a working connector, while 
```yaml
version: 0.2.0
name: cat-facts
type: http
topic: cat-facts
direction: source
parameters:
  endpoint: https://fact-apis.herokuapp.com/facts/cat
  interval: 15
  map: petfact-mapper
```
works with default parameters.

### PR DESCRIPTION
**Pull request by morenol**
Closes #285.

Also:

- Added `aggregate` support to sink connectors
- Changed sink connectors to use smart-modules by name rather than by fetching wasm file and passing again to the SPU

### CODE REVIEW

File: rust-connectors/common/src/opt.rs [hunk 1]
```diff
@@ -55,44 +58,89 @@ pub struct CommonConsumerOpt {
...
+    )]
+    pub smart_module: Option<String>,
+
+    /// (Optional) Extra input parameters passed to the smartmodule module.
+    /// They should be passed using key:value format.
+    ///
+    /// It only accepts one key:value pair. In order to pass multiple pairs, call this option multiple times.
+    ///
+    /// Example:
+    /// --smartmodule-parameters key1:value --smartmodyle-parameters key2:value
```
Comment (simlay) [inline rust-connectors/common/src/opt.rs:128]:
```suggestion
    /// --smartmodule-parameters key1:value --smartmodule-parameters key2:value
```

File: rust-connectors/common/src/opt.rs [hunk 2]
```diff
@@ -118,7 +166,9 @@ impl CommonConnectorOpt {
         let fluvio = fluvio::Fluvio::connect().await?;
         self.ensure_topic_exists().await?;
         let config_builder = TopicProducerConfigBuilder::default();
+        let params = self.smartmodule_parameters();
 
+        println!("{:?}", params);
```
Comment (simlay) [inline rust-connectors/common/src/opt.rs:171]:
```suggestion
        debug!("{:?}", params);
```

Comment (morenol) [reply rust-connectors/common/src/opt.rs]: I removed this line

**Pull request closed without merging**